cmake/OMP.cmake

```diff
@@ -0,0 +1,44 @@
+if( HAVE_OMP )
+
+    try_compile(
```

Comment:
This OMP detection mechanism is really just for OMP device offload, right? As nothing in this PR requires it, and since we're not testing it, I'd say we can drop this until we really need it and are ready support this properly.

Reply:
Note, the same applies to the source files for the detection test.